teachopencadd/talktorials/T032_compound_activity_proteochemometrics/talktorial.ipynb

```diff
@@ -0,0 +1,1923 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #45.        # Calculate model performance with regression metrics
&nbsp;&nbsp;model_performance = {}
&nbsp;&nbsp;model_performance['Pearson r'] = pearsonr(test.iloc[:, 0], predictions)[0]
&nbsp;&nbsp;model_performance['R2 score'] = r2_score(test.iloc[:, 0], predictions)
&nbsp;&nbsp;model_performance['MAE'] = mean_absolute_error(test.iloc[:, 0], predictions)
</pre><p> </p><p>Please correct me if I am overseeing something here. This bit is identical in the functions train_validate_pcm_model and train_validate_qsar_model; could we maybe add before the two functions a function called <code>_performance_metrics()</code> and use it in both model functions?</p><p> </p><p>I think the same is true for this bit:</p><pre class="ql-syntax" spellcheck="false"># Plot correlation between true and predicted values
&nbsp;&nbsp;ax = sns.scatterplot(y=test.iloc[:, 0], x=predictions, label=(f'{target} R2 = {model_performance["R2 score"]:.2f}'))
&nbsp;&nbsp;_ = sns.lineplot(x=(0, 14), y=(0, 14))
&nbsp;&nbsp;_ = ax.set_xlim((0,14))
&nbsp;&nbsp;_ = ax.set_ylim((0,14))
&nbsp;&nbsp;_ = ax.set_xlabel('Predicted')
&nbsp;&nbsp;_ = ax.set_ylabel('Observed')
</pre><p>Maybe a <code>_performance_plot()</code> function?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/teachopencadd/pull/278/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='73'/>

Reply:
<p>Very good idea :)</p>